packages/server/tests/acceptance/hbarLimiter.spec.ts

```diff
@@ -503,6 +499,186 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
+            });
+
+            it('Should successfully add spending plans', async () => {
+              const plan = await ethAddressSpendingPlanRepository.findByAddress(
+                priviledgedEvmAccount.address,
+                requestDetails,
+              );
+
+              expect(plan.ethAddress).to.eq(priviledgedEvmAccount.address);
+              expect(plan.planId).to.eq(privilegedHbarSpendingPlan.id);
+
+              const spendingPlan = await hbarSpendingPlanRepository.findByIdWithDetails(plan.planId, requestDetails);
+              expect(spendingPlan.active).to.be.true;
+              expect(spendingPlan.amountSpent).to.eq(0);
+              expect(spendingPlan.subscriptionTier).to.eq(SubscriptionTier.PRIVILEGED);
+            });
+
+            it('Should increase the amount spent by the privileged spending plan', async () => {
+              const contract = await deployContract(largeContractJson, priviledgedEvmAccount.wallet);
+
+              // awaiting for HBAR limiter to finish updating expenses in the background
+              await Utils.wait(6000);
+
+              const spendingPlan = await hbarSpendingPlanRepository.findByIdWithDetails(
+                privilegedHbarSpendingPlan.id,
+                requestDetails,
+              );
+
+              const expectedCost = await getExpectedCostOfLastLargeTx(contract.deploymentTransaction()!.data);
+              const amountSpent = spendingPlan.amountSpent;
+
+              expect(amountSpent).to.be.approximately(expectedCost, 0.009 * expectedCost);
+            });
+
+            it('Should eventually exhaust the hbar limit for PRIVILEDGED user', async () => {
+              const fileChunkSize = Number(ConfigService.get('FILE_APPEND_CHUNK_SIZE')) || 5120;
+              const exchangeRateResult = (await mirrorNode.get(`/network/exchangerate`, requestId)).current_rate;
+              const exchangeRateInCents = exchangeRateResult.cent_equivalent / exchangeRateResult.hbar_equivalent;
+              const remainingHbarsBefore = Number(await metrics.get(testConstants.METRICS.REMAINING_HBAR_LIMIT));
+
+              const factory = new ethers.ContractFactory(
+                largeContractJson.abi,
+                largeContractJson.bytecode,
+                accounts[0].wallet,
+              );
+              const deployedTransaction = await factory.getDeployTransaction();
+              const estimatedTxFee = estimateFileTransactionsFee(
+                deployedTransaction.data.length,
+                fileChunkSize,
+                exchangeRateInCents,
+              );
+
+              try {
+                for (let i = 0; i < 50; i++) {
+                  await deployContract(largeContractJson, priviledgedEvmAccount.wallet);
+                }
+                expect.fail(`Expected an error but nothing was thrown`);
+              } catch (e: any) {
+                expect(e.message).to.contain(predefined.HBAR_RATE_LIMIT_EXCEEDED.message);
```

Comment:
we should verify here that `i` is greater than the `<expectedAmountOfLargeContractsWhichCanBeSubmittedByPrivilegedTier>`

Reply:
Are you suggesting we add the `expectedAmountOfLargeContractsWhichCanBeSubmittedByPrivilegedTier` constant?

One concern is that the number of large contract deployments depends on the `HBAR_RATE_LIMIT_PRIVILEGED`, which is currently set to a lower value (50 HBAR) for testing purposes. Therefore, the value of `expectedAmountOfLargeContractsWhichCanBeSubmittedByPrivilegedTier` might not be accurate in this context.

Reply:
No, more like a dynamically calculated value based on the estimated tx fee and `HBAR_RATE_LIMIT_PRIVILEGED`:

```typescript
const largeContractDeployCost = estimateFileTransactionsFee(
  contract.deploymentTransaction()!.data.length,
  fileChunkSize,
  exchangeRateInCents
);
const expectedAmountOfDeploys = Math.floor(HBAR_RATE_LIMIT_PRIVILEGED / largeContractDeployCost);

let i = 0;
try {
  for (i = 0; i < expectedAmountOfDeploys; i++) {
    ...
  }
} catch (e: any) {
  expect(e.message).to.contain(predefined.HBAR_RATE_LIMIT_EXCEEDED.message);
  expect(i).to.be.eq(expectedAmountOfDeploys);
  ...
}
```

Reply:
It would be nice to have this for other tests as well, because we want to confirm not only that at some point they reach the limit but also after how many deploys they have reached it (this will verify that the tier limit we have set is being met)

Reply:
This is a very good point thanks for the suggestion! Updated the tests.